tests/test_github_client.py

```
@@ -0,0 +1,417 @@
+
+    @pytest.fixture
+    def client(self, mock_github):
+        """Create a GitHubClient instance with mocked PyGithub."""
+        with patch('gh_pr.core.github.Github'):
+            return GitHubClient("test_token")
+
+    def test_init(self):
+        """Test GitHubClient initialization."""
+        client = GitHubClient("test_token_123")
+        # Token should be stored privately
+        assert client._token == "test_token_123"
```

Ruff (0.13.1)

Possible hardcoded password assigned to: "_token"

(S105)